src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -122,15 +137,112 @@ protected ExternalWorkspace run() throws Exception {
     }
 
     /**
+     * Iterates through given disk pool list and finds the {@link DiskPool} that has the {@link DiskPool#diskPoolId}
+     * equal to the given disk pool id parameter.
+     *
+     * @param diskPoolId the matching id that the disk pool should have
+     * @param diskPools  the list of disk pools
+     * @return the disk pool whose {@link DiskPool#diskPoolId} is equal to the given disk pool id parameter
+     * @throws IOException if there isn't find any disk pool matching the disk pool id, or
+     *                     if the disk pool doesn't have defined any {@link Disk} entries
+     */
+    @Nonnull
+    private static DiskPool findDiskPool(@Nonnull String diskPoolId, @Nonnull List<DiskPool> diskPools) throws IOException {
```

Comment:
Fine for the private method. For public API a nullable method would be preferable